charts/cluster-security/templates/bindings.yaml

```diff
@@ -1,30 +1,38 @@
-# This role binding allows all pods in the 'default' and 'apps' namespace to use the baseline PSP.
-apiVersion: rbac.authorization.k8s.io/v1
+ {{- range .Values.baselineNamespaces }}
 kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
 metadata:
-  name: psp-baseline-rolebinding
-  namespace: apps
-roleRef:
-  apiGroup: rbac.authorization.k8s.io
-  kind: ClusterRole
   name: psp-baseline
+  namespace: {{ . }}
 subjects:
 - kind: Group
-  name: system:authenticated
-  namespace: apps
+  name: system:serviceaccounts
```

Comment:
It seems odd to be specifying `system:serviceaccounts` (i.e. all service accounts in all namespaces) in a RoleBinding, which is inherently specific to a namespace. Is that intended? Did you mean to include the namespace here?

Reply:
Alternatively it might be simpler to use ClusterRoleBinding for some of this.

Reply:
From [this](https://kubernetes.io/docs/concepts/security/pod-security-policy/),  it advises using a rolebinding to grant usage for pods being run in the same namespace as the binding. When I tested with a clusterolebinding it didn't work properly. I'll drop the namespace argument here in the subjects since it's specified in the metadata of the rolebinding.

Reply:
It should work with either - the docs that you linked to have examples with RoleBinding and ClusterRoleBinding.

Reply:
The clusterrolebinding looks straightforward enough, but when it came to testing whether the psp would restrict deployments of say for example a root container deployment, it just wouldn't work.
Did some reading and its fine to apply regular RoleBindings to ClusterRoles.
I can look to testing clusterrolebindings again, but at the moment the regular rolebindings work.

Reply:
Like I said, either should work. I don't mind whether we use ClusterRoleBinding or RoleBinding here; I'm just trying to ensure that the policy intent is clear. So if we have a RoleBinding it'd make sense to have the subject be `system:serviceaccounts:{{ . }}` explicitly (i.e. the namespace's service accounts) rather than specifying all service accounts but then relying on the fact that it'll be implicitly restricted to principles in the namespace of the RoleBinding. Just makes it that much easier to understand - and by gosh this area is complicated enough already 😅